[docs] Fix broken link & update MUI packages explanation

siriwatknp opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

close #29493

**Preview**: https://deploy-preview-29583--material-ui.netlify.app/guides/understand-mui-packages/#main-content

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

siriwatknp commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,42 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
 The packages can be categorized into 3 layers, as shown in the picture below:
 
-<img src="/static/images/packages/mui-packages.png" style="width: 700px; max-width: 100%;" />
+<img src="/static/images/packages/mui-packages.png" style="display: block; width: 448px; max-width: 100%; margin: auto;" />
```
remove spacing in the image to make it looks good in mobile.

oliviertassinari replied:
This style seems to already comes by default

```suggestion
<img src="/static/images/packages/mui-packages.png" style="display: block; width: 448px; margin: auto;" />
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
 - Components created with `styled` are themeable via slots & variants.
 
-> **Note**: you will have to install either `emotion` or `styled-components`, because the `system` package depends on it.
+> **Note**: you will have to install either `emotion` or `styled-components`, because the styled-engine package depends on it.
```
```suggestion
> **Note**: you will have to install either `emotion` or `styled-components`, because the respective styled-engine package depends on it.
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
 
+> 💡 Glossary
+>
+> **install** refers to the action when you run `yarn add $module` or `npm install $module`.
```
```suggestion
> **install** refers to running `yarn add $module` or `npm install $module`.
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
+>
+> **install** refers to the action when you run `yarn add $module` or `npm install $module`.
+>
+> **import** means you want to use some APIs from the module by writing `import ... from '$module'`.
```
```suggestion
> **import** refers to making a module API available in your code by adding `import ... from '$module'`.
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
-This layer is specifically related to stylesheets.
-Currently we use `emotion` as the default style library for creating stylesheets.
-In addition, we also provide an adapter for developers who prefer to use styled-components.
+This layer is specifically related to stylesheet generation (CSS-in-js). We have introduced new styled-engines in v5 to unlock more possibilities and enable customization enhancement. Usually, developers does not need to interact with this layer on a daily basis because it is used internally in `@mui/system`.
```
```suggestion
This layer is specifically related to stylesheet generation (CSS-in-JS). We have introduced new styled-engines in v5 to unlock more possibilities and enable enhanced customization. Usually, developers does not need to interact with this layer on a daily basis because it is used internally in `@mui/system`.
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
+They come in two packages:
 
-These are the packages in this layer:
+- `@mui/styled-engine` : an [emotion](https://emotion.sh/docs/styled) wrapper
```
```suggestion
- `@mui/styled-engine` : an [emotion](https://emotion.sh/docs/styled) wrapper.
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
 
-These are the packages in this layer:
+- `@mui/styled-engine` : an [emotion](https://emotion.sh/docs/styled) wrapper
+- `@mui/styled-engine-sc` : a [styled-components](https://styled-components.com/docs/basics#getting-started) wrapper
```
```suggestion
- `@mui/styled-engine-sc` : a [styled-components](https://styled-components.com/docs/basics#getting-started) wrapper.
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
-- `@mui/styled-engine` : an emotion wrapper
-- `@mui/styled-engine-sc` : a styled-components wrapper
-- `@mui/styles` : JSS wrapper (`deprecated`)
+The reason that we create these adapters is to unify the APIs to support both `emotion` and `styled-components`, so that developers can choose what suit them the most.
```
```suggestion
These adapters unify the APIs of both `emotion` and `styled-components`, so that developers can choose to use whichever suits them best.
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
+The reason that we create these adapters is to unify the APIs to support both `emotion` and `styled-components`, so that developers can choose what suit them the most.
 
-> These packages are already included in the design system package, so you don't need to worry about _importing_ them if you're using `@mui/material`, but you will need to [install](`@mui/styled-engine-sc`) if you wish to replace emotion as the default, as well as your chosen style library.
+The old styled-engine `@mui/styles` (JSS wrapper) is deprecated and will be removed in the future.
```
```suggestion
The previous style library `@mui/styles` (JSS wrapper) is deprecated and will be removed in the future.
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
-> These packages are already included in the design system package, so you don't need to worry about _importing_ them if you're using `@mui/material`, but you will need to [install](`@mui/styled-engine-sc`) if you wish to replace emotion as the default, as well as your chosen style library.
+The old styled-engine `@mui/styles` (JSS wrapper) is deprecated and will be removed in the future.
+
+> The details about changing styling solution is in [this RFC](https://github.com/mui-org/material-ui/issues/22342)
```
```suggestion
> The details about the change in styling solution are in [this RFC](https://github.com/mui-org/material-ui/issues/22342).
```

mbrookes commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
 There is only one package in this layer - `@mui/system`.
-It uses a styled-engine package and provides APIs for building a design system from scratch, for example, `styled` is enhanced to provide more theming capabilities.
+
+It uses emotion adapter (`@mui/styled-engine`) as a default styled-engine to create APIs for building a design system from scratch. For example, [`styled`](/system/styled/#main-content) from the styled-engine is enhanced to provide more theming capabilities.
```
```suggestion
It uses the emotion adapter (`@mui/styled-engine`) as the default styled-engine to create APIs for building a design system from scratch. For example, [`styled`](/system/styled/#main-content) from the styled-engine is enhanced to provide more theming capabilities.
```

mbrookes reviewed (changes requested):
`styled-engine` is the (unfortunate) name for the style library wrapper / adapter package, but there's no such thing as a "styled-engine" as a generic term. Should we stick to something like "style library adapter" when not referring to the `styled-engine` package specifically? It's quite confusing otherwise.

mnajdova commented:
> Should we stick to something like "style library adapter" when not referring to the styled-engine package specifically? It's quite confusing otherwise.

"style library adapter" sounds good to me.

oliviertassinari commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
 
+> 💡 Glossary
+>
+> **install** refers to running `yarn add $module` or `npm install $module`.
+>
+> **import** refers to making a module API available in your code by adding `import ... from '$module'`.
```
Would have a ul > li here help? I personally got confused the first time I read this section:

<img width="451" alt="Screenshot 2021-11-10 at 12 01 29" src="https://user-images.githubusercontent.com/3165635/141101526-24b6d19e-b050-4d29-aaf1-80262ded0f2b.png">

siriwatknp replied:
👍

oliviertassinari commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
-This layer is specifically related to stylesheets.
-Currently we use `emotion` as the default style library for creating stylesheets.
-In addition, we also provide an adapter for developers who prefer to use styled-components.
+This layer is specifically related to stylesheet generation (CSS-in-JS). We have introduced new styled-engines in v5 to unlock more possibilities and enable enhanced customization. Usually, developers does not need to interact with this layer on a daily basis because it is used internally in `@mui/system`.
```
Prefer one line per sentence. Also, I changed does -> do, we -> MUI

```suggestion
This layer is specifically related to stylesheet generation (CSS-in-JS).
MUI has introduced new styled-engines in v5 to unlock more possibilities and enable enhanced customization.
Usually, developers do not need to interact with this layer on a daily basis because it is used internally in `@mui/system`.
```

oliviertassinari commented on docs/src/pages/guides/understand-mui-packages/understand-mui-packages.md:
```diff
@@ -34,35 +34,48 @@ For this reason, abstracting into smaller packages not only allows MUI to grow o
+
+If you want to switch the styled-engine to use styled-components, [follow this guide](/guides/styled-engine/#how-to-switch-to-styled-components).
+
+<img src="/static/images/packages/mui-system.png" style="display: block; width: 720px; max-width: 100%; margin: 40px auto;" />
```
I believe it comes by default.

```suggestion
<img src="/static/images/packages/mui-system.png" style="display: block; width: 720px; margin: 40px auto;" />
```

siriwatknp commented:
> Is this capitalization correct?

changed to TL;DR

> This is not how we have been treating them so far

removed.

> core -> base

updated the image.

> This can be controversial

removed.

mnajdova reviewed (approved):
LGTM

oliviertassinari commented:
Are we sure about the size of the images?

For example, /static/images/packages/mui-packages.png displays blurry on my x2 screen. Its displayed size is 448px so I would assume that we can use an image with an intrinsic size of 896px (x2). It's currently 1381px. Same for docs/public/static/images/packages/mui-system.png

siriwatknp commented:
> Are we sure about the size of the images?
> 
> For example, /static/images/packages/mui-packages.png displays **blurry** on my x2 screen. Its displayed size is 448px so I would assume that we can use an image with an intrinsic size of 896px (x2). It's currently 1381px. Same for docs/public/static/images/packages/mui-system.png. It definitely makes a visible difference in my screen display environment, but I don't know if it would also help developers with an x1 display.
> 
> BTW, I have purchased the latest MackBook Pro, only for the XDR display, so I can be pickier in the future with visual aesthetics 👼. Looks good otherwise

The size looks correct to me, 1381px is more than 896px so wouldn't it give a similar result.

I can open a fix later.

danilo-leal commented:
> so I can be pickier in the future with visual aesthetics 👼. 

👀 👀 👀 fancy!

oliviertassinari commented:
>  1381px is more than 896px so wouldn't it give a similar result.

@siriwatknp Yeah, I don't know if this only helps people like me on Apple hardware or Windows developers with regular screens too. In the case of Apple, the CSS pixel is 448, so it uses 896 real pixels. Now, if the image is also 896 pixels, then the OS doesn't need to resize it 👌. With a larger image size, it resizes it, the **text** loses its crispness, it becomes fuzzy, harder to read.
